numba/typed/typedlist.py

```diff
@@ -356,13 +356,24 @@ def pop(self, i=-1):
 
     def extend(self, iterable):
         if not self._typed:
+            iterable_length = len(iterable)
+            # Empty iterable, do nothing
+            if iterable_length == 0:
+                return self
             # Need to get the first element of the iterable to initialise the
             # type of the list. FIXME: this may be a problem if the iterable
             # can not be sliced.
             self._initialise_list(iterable[0])
             self.append(iterable[0])
-            return _extend(self, iterable[1:])
-        return _extend(self, iterable)
+            # Iterable has more values to extend from
+            if iterable_length > 1:
+                return _extend(self, iterable[1:])
+            else:
+                return self
```

Comment:
Could this just be:
```
            self._initialise_list(iterable[0])
            return _extend(self, iterable)
```
i.e. init type, then extend, is there a need to append then slice from 1?

Reply:
Yes, this can be significantly simplified.